oauthenticator/mediawiki.py

```diff
@@ -116,6 +120,23 @@ class MWOAuthenticator(OAuthenticator):
         help='Full path to index.php of the MW instance to use to log in'
     )
 
+    pass_secrets = Bool(
+        False,
+        config=True,
+        help='Pass OAuth consumer and access secrets to the spawner'
+    )
+
+    def pre_spawn_start(self, user, spawner):
+        if not self.pass_secrets:
+            return
+
+        spawner.env.update({
+            'CLIENT_SECRET': self.client_secret,
+            'CLIENT_ID': self.client_id,
+            'ACCESS_KEY': user.state['ACCESS_KEY'],
```

Comment:
Spawners shouldn't ever access user.state. These should always be loaded by `self.load_state` and retrieved by `self.get_state`.